engine/src/player-platform.cpp

```diff
@@ -1823,6 +1855,8 @@ IO_stat MCPlayer::save(IO_handle stream, uint4 p_part, bool p_force_ext)
 			return stat;
 		if ((stat = IO_write_string(filename, stream)) != IO_NORMAL)
 			return stat;
+        if ((stat = IO_write_string(resolved_filename, stream)) != IO_NORMAL)
```

Comment:
You can't do this as it changes the file format!

The player object should save 'filename' as what was set as the property; but use the 'resolved' filename at the point the movie is prepared.